src/gui/mainwindow.cpp

```diff
@@ -1742,30 +1743,30 @@ void MainWindow::on_actionRSSReader_triggered()
 void MainWindow::on_actionSearchWidget_triggered()
 {
     if (!m_hasPython && m_ui->actionSearchWidget->isChecked()) {
-        int pythonVersion = Utils::Misc::pythonVersion();
+        int majorVersion = Utils::Python::majorVersion();
 
         // Check if python is already in PATH
-        if (pythonVersion > 0)
+        if (majorVersion > 0)
             // Prevent translators from messing with PATH
             Logger::instance()->addMessage(tr("Python found in %1: %2", "Python found in PATH: /usr/local/bin:/usr/bin:/etc/bin")
                 .arg("PATH", qgetenv("PATH").constData()), Log::INFO);
 #ifdef Q_OS_WIN
         else if (addPythonPathToEnv())
-            pythonVersion = Utils::Misc::pythonVersion();
+            majorVersion = Utils::Python::majorVersion();
 #endif
 
         bool res = false;
 
-        if ((pythonVersion == 2) || (pythonVersion == 3)) {
+        if ((majorVersion == 2) || (majorVersion == 3)) {
             // Check Python minimum requirement: 2.7.9 / 3.3.0
-            QString version = Utils::Misc::pythonVersionComplete();
+            QString version = Utils::Python::fullVersion();
             QStringList splitted = version.split('.');
             if (splitted.size() > 2) {
                 int middleVer = splitted.at(1).toInt();
                 int lowerVer = splitted.at(2).toInt();
-                if (((pythonVersion == 2) && (middleVer < 7))
-                    || ((pythonVersion == 2) && (middleVer == 7) && (lowerVer < 9))
-                    || ((pythonVersion == 3) && (middleVer < 3))) {
+                if (((majorVersion == 2) && (middleVer < 7))
```

Comment:
It seems to me that these conditional expressions would look simpler as follows:
```c++
// a bit above
auto pythonVersion = Utils::Python::detectedVersion(); // return Utils::Version
// and here
if (pythonVersion < {2,7,9}
  || (pythonVersion.major() == 3 && pythonVersion < {3,3,0})
```

Reply:
same here, I intend to keep this PR small and hopefully get merged earlier.